### ISSUE TITLE
Add precise language on DID unique suffix generation and test vector 

### ISSUE THREAD

**Issue by thehenrytsai** [decentralized-identity/sidetree#550]
Example test vectors can be found at: https://tools.ietf.org/html/rfc3394

**Comment by troyronda** [comment]
Related note:

https://identity.foundation/sidetree/docs/spec/#did-suffix-composition refers to the
> Create operation’s Suffix Data Object

but this object isn't defined in that section.

**Comment by troyronda** [comment]
Another note: multi-hash isn't mentioned in the spec.

**Comment by csuwildcat** [comment]
**WIP:**

DID Methods based on the Sidetree protocol all share the same identifier format. The unique identifier string is a hash of a set of initialization values the controller generates when they create a Sidetree-based DID. This composite hash of initial state values enables Sidetree DIDs to be _self-certifying_, meaning the person or entity who generates a Sidetree-based DID knows their unique identifier immediately, and it is cryptographic secured for instant use (for more on instant use capabilities of Sidetree DIDs see the [Long-Form Sidetree Format](#) section below).

The process below MUST be used to correctly generate a Sidetree-based DID identifier string:

1. From the values generated during the [Create](#create) operation process, assemble an [_Anchor File Create Entry_](#anchor-file-create-entry) object, composed as follows:

    ```
    {
      "delta_hash": DELTA_HASH,
      "recovery_key": JWK_OBJECT,
      "recovery_commitment": COMMITMENT_HASH
    }
    ```
    - The object MUST contain a `delta_hash` property, and its value MUST be a hash of the [Create Operation Delta Object](create-data-object).
    - The object MUST contain a `recovery_key` property, and its value MUST be the _Initial Recovery Public Key_.
    - The object MUST contain a `recovery_commitment` property, and its value MUST be the _Initial Recovery Commitment_

**Comment by OR13** [comment]
suggested change:

HTTP POST http://example.com/operations { body } -H { accept: application/json }

body : {
 type: 'create',
suffix_data: base64url ( {
  "delta_hash": DELTA_HASH,
  "recovery_key": JWK_OBJECT,
  "recovery_commitment": COMMITMENT_HASH
})
delta: base64url({
 "patches": PATCH_ARRAY,
  "update_commitment": COMMITMENT_HASH
})

}

Suffix = ASCII( Multihash( SHA256,  suffix_data ) )

**Comment by troyronda** [comment]
Recommend a note that JSON Canonicalization MAY be performed: https://github.com/decentralized-identity/sidetree/issues/564

**Comment by jmandel** [comment]
I agree this clarification is important to align with the current implementation.

I'm curious whether there's an underlying design reason that the hash is calculated over the stringified JSON though, instead of calculated over the base64url-encoded value that's sent over the wire?

**Comment by OR13** [comment]
@csuwildcat @thehenrytsai to define what happens in the spec language.

Assuming this is documented, we can have test vectors.

**Comment by OR13** [comment]
Blocked pending reference implementation catchup...

**Comment by csuwildcat** [comment]
Henry, are the vectors the same as in the spec here?: https://identity.foundation/sidetree/spec/#long-form-did-uris

**Comment by csuwildcat** [comment]
@thehenrytsai are the vectors present in the spec, as linked above, true and correct for what is implemented?

**Comment by csuwildcat** [comment]
ACTION: Dan will update examples

**Comment by troyronda** [comment]
ACTION: Copy reference implementation test vectors into spec.
@thehenrytsai @csuwildcat

**Comment by troyronda** [comment]
Copying the resolution output test vectors into the Spec has been done.
